src/client/pfs.go

```diff
@@ -1872,7 +1872,7 @@ func (w *putObjWriteCloser) Write(p []byte) (int, error) {
 }
 
 func (w *putObjWriteCloser) Close() error {
-	if w.request.Obj == "" {
+	if w.request.Obj != "" {
```

Comment:
I'm a little unsure if this change is correct. Based on the comment below it looks like this line is supposed to run when the block is empty, which I think is the same as when `Obj == ""`. Not sure though, I'd defer to @brycemcanally on this.

Reply:
To be clear, this `putObjWriteCloser` is only used for the `PutObjDirect` RPC, which until 1.11.5 was not used anywhere in the codebase (it was used a bit in the worker before the refactor, I believe, but that went away until I changed the job artifact object usage in 1.11.5).  `Obj == ""` appears to only be checking if a previous chunk of data has been sent.  In practice, this logic being wrong meant that we would send an extra unused message at the end of the stream, and writing an empty object would error.  I've added another test that reproduces this with an empty object on the old code and works fine with the new code.

Reply:
Ahh, that makes sense.